server/src/router/index.js

```diff
@@ -28,23 +30,17 @@ router.get('/:categoryId/courses', getCatcourses);
 router.get('/courses/:courseId', getCourseDetails);
 router.get('/topCourses', getTopRatedCourses);
 
-router.all(
-  [
-    '/favorite',
-    '/favorite/add/:courseId',
-    '/favorite/delete/:courseId',
-    '/auth',
-  ],
-  verifyUser
-);
+router.all(['/favorite', '/favorite/:id', '/comment/:id', '/auth'], verifyUser);
```

Comment:
Fix protected routes : 
```suggestion
router.all(['/favorite', '/favorite/:courseId', '/comment/:commentId', '/comment/:courseId', '/auth'], verifyUser);
```